In [1796]:
import pandas as pd
import numpy as np

In [1797]:
df=pd.read_csv('Combined_News_DJIA.csv')

In [1798]:
df

Date  Label                                               Top1  \
0     2008-08-08      0  b"Georgia 'downs two Russian warplanes' as cou...   
1     2008-08-11      1  b'Why wont America and Nato help us? If they w...   
2     2008-08-12      0  b'Remember that adorable 9-year-old who sang a...   
3     2008-08-13      0  b' U.S. refuses Israel weapons to attack Iran:...   
4     2008-08-14      1  b'All the experts admit that we should legalis...   
...          ...    ...                                                ...   
1984  2016-06-27      0  Barclays and RBS shares suspended from trading...   
1985  2016-06-28      1  2,500 Scientists To Australia: If You Want To ...   
1986  2016-06-29      1                   Explosion At Airport In Istanbul   
1987  2016-06-30      1  Jamaica proposes marijuana dispensers for tour...   
1988  2016-07-01      1  A 117-year-old woman in Mexico City finally re...   

                                                   Top2  \
0               b'BREAKING: Musharraf to be impeached.'   
1           b'Bush puts foot down on Georgian conflict'   
2                    b"Russia 'ends Georgia operation'"   
3     b"When the president ordered to attack Tskhinv...   
4     b'War in South Osetia - 89 pictures made by a ...   
...                                                 ...   
1984  Pope says Church should ask forgiveness from g...   
1985  The personal details of 112,000 French police ...   
1986  Yemeni former president: Terrorism is the offs...   
1987  Stephen Hawking says pollution and 'stupidity'...   
1988   IMF chief backs Athens as permanent Olympic host   

                                                   Top3  \
0     b'Russia Today: Columns of troops roll into So...   
1     b"Jewish Georgian minister: Thanks to Israeli ...   
2     b'"If we had no sexual harassment we would hav...   
3     b' Israel clears troops who killed Reuters cam...   
4     b'Swedish wrestler Ara Abrahamian throws away ...   
...                                                 ...   
1984  Poland 'shocked' by xenophobic abuse of Poles ...   
1985  S&amp;P cuts United Kingdom sovereign credit r...   
1986  UK must accept freedom of movement to access E...   
1987  Boris Johnson says he will not run for Tory pa...   
1988  The president of France says if Brexit won, so...   

                                                   Top4  \
0     b'Russian tanks are moving towards the capital...   
1     b'Georgian army flees in disarray as Russians ...   
2     b"Al-Qa'eda is losing support in Iraq because ...   
3     b'Britain\'s policy of being tough on drugs is...   
4     b'Russia exaggerated the death toll in South O...   
...                                                 ...   
1984  There will be no second referendum, cabinet ag...   
1985                Huge helium deposit found in Africa   
1986  Devastated: scientists too late to captive bre...   
1987  Six gay men in Ivory Coast were abused and for...   
1988  British Man Who Must Give Police 24 Hours' Not...   

                                                   Top5  \
0     b"Afghan children raped with 'impunity,' U.N. ...   
1         b"Olympic opening ceremony fireworks 'faked'"   
2     b'Ceasefire in Georgia: Putin Outmaneuvers the...   
3     b'Body of 14 year old found in trunk; Latest (...   
4     b'Missile That Killed 9 Inside Pakistan May Ha...   
...                                                 ...   
1984      Scotland welcome to join EU, Merkel ally says   
1985  CEO of the South African state broadcaster qui...   
1986  British Labor Party leader Jeremy Corbyn loses...   
1987  Switzerland denies citizenship to Muslim immig...   
1988  100+ Nobel laureates urge Greenpeace to stop o...   

                                                   Top6  \
0     b'150 Russian tanks have entered South Ossetia...   
1     b'What were the Mossad with fraudulent New Zea...   
2     b'Why Microsoft and Intel tried to kill the XO...   
3  

In [1799]:
df.isnull().sum()

Date     0
Label    0
Top1     0
Top2     0
Top3     0
Top4     0
Top5     0
Top6     0
Top7     0
Top8     0
Top9     0
Top10    0
Top11    0
Top12    0
Top13    0
Top14    0
Top15    0
Top16    0
Top17    0
Top18    0
Top19    0
Top20    0
Top21    0
Top22    0
Top23    1
Top24    3
Top25    3
dtype: int64

# As we could see we have very few records as null in our data so we will drop it

In [1800]:
df.dropna(inplace=True)

In [1801]:
df.shape


(1986, 27)

In [1802]:
df['Label'].value_counts()

1    1062
0     924
Name: Label, dtype: int64

# Pre Processing the data

In [1803]:
# Removing the puntuations
Data=df.iloc[:,2:27]
Data.replace("[^a-zA-Z]"," ",regex=True,inplace=True)

In [1804]:
# Renaming the columns just for the ease

In [1805]:
lst= [i for i in range(25)]
new_index=[str(i) for i in lst]
Data.columns=new_index
Data.head()

0  \
0  b Georgia  downs two Russian warplanes  as cou...   
1  b Why wont America and Nato help us  If they w...   
2  b Remember that adorable   year old who sang a...   
3  b  U S  refuses Israel weapons to attack Iran ...   
4  b All the experts admit that we should legalis...   

                                                   1  \
0            b BREAKING  Musharraf to be impeached     
1        b Bush puts foot down on Georgian conflict    
2                 b Russia  ends Georgia operation     
3  b When the president ordered to attack Tskhinv...   
4  b War in South Osetia      pictures made by a ...   

                                                   2  \
0  b Russia Today  Columns of troops roll into So...   
1  b Jewish Georgian minister  Thanks to Israeli ...   
2  b  If we had no sexual harassment we would hav...   
3  b  Israel clears troops who killed Reuters cam...   
4  b Swedish wrestler Ara Abrahamian throws away ...   

                                                   3  \
0  b Russian tanks are moving towards the capital...   
1  b Georgian army flees in disarray as Russians ...   
2  b Al Qa eda is losing support in Iraq because ...   
3  b Britain  s policy of being tough on drugs is...   
4  b Russia exaggerated the death toll in South O...   

                                                   4  \
0  b Afghan children raped with  impunity   U N  ...   
1      b Olympic opening ceremony fireworks  faked     
2  b Ceasefire in Georgia  Putin Outmaneuvers the...   
3  b Body of    year old found in trunk  Latest  ...   
4  b Missile That Killed   Inside Pakistan May Ha...   

                                                   5  \
0  b     Russian tanks have entered South Ossetia...   
1  b What were the Mossad with fraudulent New Zea...   
2  b Why Microsoft and Intel tried to kill the XO...   
3  b China has moved     million  quake survivors...   
4  b Rushdie Condemns Random House s Refusal to P...   

                                                   6  \
0  b Breaking  Georgia invades South Ossetia  Rus...   
1  b Russia angered by Israeli military sale to G...   
2  b Stratfor  The Russo Georgian War and the Bal...   
3  b Bush announces Operation Get All Up In Russi...   
4  b Poland and US agree to missle defense deal  ...   

                                                   7  \
0  b The  enemy combatent  trials are nothing but...   
1  b An American citizen living in S Ossetia blam...   
2  b I m Trying to Get a Sense of This Whole Geor...   
3             b Russian forces sink Georgian ships     
4  b Will the Russians conquer Tblisi  Bet on it ...   

                                                   8  \
0  b Georgian troops retreat from S  Osettain cap...   
1  b Welcome To World War IV  Now In High Definit...   
2  b The US military was surprised by the timing ...   
3  b The commander of a Navy air reconnaissance s...   
4  b Russia exaggerating South Ossetian death tol...   

                                                   9  ...  \
0  b Did the U S  Prep Georgia for War with Russia    ...   
1  b Georgia s move  a mistake of monumental prop...  ...   
2    b U S  Beats War Drum as Iran Dumps the Dollar   ...   
3  b     of CNN readers  Russia s actions in Geor...  ...   
4  b  Musharraf expected to resign rather than fa...  ...   

                                                  15  \
0  b Georgia Invades South Ossetia   if Russia ge...   
1  b Israel and the US behind the Georgian aggres...   
2  b U S  troops still in Georgia  did you know t...   
3                      b Elephants extinct by          
4  b Bank analyst forecast Georgian crisis   days...   

                                                  16  \
0                b Al Qaeda Faces Islamist Backlash    
1  b  Do not believe TV  neither Russian nor Geor...   
2       b Why Russias response to Georgia was right    
3  b US humanitarian missions soon in Georgia   i...   
4  b Georgia confict could set bac

# Converting all the characters into a lower character


In [1806]:
for index in new_index:
    Data[index]=Data[index].str.lower()

Data.head()

0  \
0  b georgia  downs two russian warplanes  as cou...   
1  b why wont america and nato help us  if they w...   
2  b remember that adorable   year old who sang a...   
3  b  u s  refuses israel weapons to attack iran ...   
4  b all the experts admit that we should legalis...   

                                                   1  \
0            b breaking  musharraf to be impeached     
1        b bush puts foot down on georgian conflict    
2                 b russia  ends georgia operation     
3  b when the president ordered to attack tskhinv...   
4  b war in south osetia      pictures made by a ...   

                                                   2  \
0  b russia today  columns of troops roll into so...   
1  b jewish georgian minister  thanks to israeli ...   
2  b  if we had no sexual harassment we would hav...   
3  b  israel clears troops who killed reuters cam...   
4  b swedish wrestler ara abrahamian throws away ...   

                                                   3  \
0  b russian tanks are moving towards the capital...   
1  b georgian army flees in disarray as russians ...   
2  b al qa eda is losing support in iraq because ...   
3  b britain  s policy of being tough on drugs is...   
4  b russia exaggerated the death toll in south o...   

                                                   4  \
0  b afghan children raped with  impunity   u n  ...   
1      b olympic opening ceremony fireworks  faked     
2  b ceasefire in georgia  putin outmaneuvers the...   
3  b body of    year old found in trunk  latest  ...   
4  b missile that killed   inside pakistan may ha...   

                                                   5  \
0  b     russian tanks have entered south ossetia...   
1  b what were the mossad with fraudulent new zea...   
2  b why microsoft and intel tried to kill the xo...   
3  b china has moved     million  quake survivors...   
4  b rushdie condemns random house s refusal to p...   

                                                   6  \
0  b breaking  georgia invades south ossetia  rus...   
1  b russia angered by israeli military sale to g...   
2  b stratfor  the russo georgian war and the bal...   
3  b bush announces operation get all up in russi...   
4  b poland and us agree to missle defense deal  ...   

                                                   7  \
0  b the  enemy combatent  trials are nothing but...   
1  b an american citizen living in s ossetia blam...   
2  b i m trying to get a sense of this whole geor...   
3             b russian forces sink georgian ships     
4  b will the russians conquer tblisi  bet on it ...   

                                                   8  \
0  b georgian troops retreat from s  osettain cap...   
1  b welcome to world war iv  now in high definit...   
2  b the us military was surprised by the timing ...   
3  b the commander of a navy air reconnaissance s...   
4  b russia exaggerating south ossetian death tol...   

                                                   9  ...  \
0  b did the u s  prep georgia for war with russia    ...   
1  b georgia s move  a mistake of monumental prop...  ...   
2    b u s  beats war drum as iran dumps the dollar   ...   
3  b     of cnn readers  russia s actions in geor...  ...   
4  b  musharraf expected to resign rather than fa...  ...   

                                                  15  \
0  b georgia invades south ossetia   if russia ge...   
1  b israel and the us behind the georgian aggres...   
2  b u s  troops still in georgia  did you know t...   
3                      b elephants extinct by          
4  b bank analyst forecast georgian crisis   days...   

                                                  16  \
0                b al qaeda faces islamist backlash    
1  b  do not believe tv  neither russian nor geor...   
2       b why russias response to georgia was right    
3  b us humanitarian missions soon in georgia   i...   
4  b georgia confict could set bac

In [1807]:
Data.shape

(1986, 25)

# Combining all the top 25 news into 1 paragraph

In [1808]:
' '.join(str(x) for x in Data.iloc[1,0:25])

'b why wont america and nato help us  if they wont help us now  why did we help them in iraq   b bush puts foot down on georgian conflict  b jewish georgian minister  thanks to israeli training  we re fending off russia   b georgian army flees in disarray as russians advance   gori abandoned to russia without a shot fired  b olympic opening ceremony fireworks  faked   b what were the mossad with fraudulent new zealand passports doing in iraq   b russia angered by israeli military sale to georgia  b an american citizen living in s ossetia blames u s  and georgian leaders for the genocide of innocent people  b welcome to world war iv  now in high definition   b georgia s move  a mistake of monumental proportions   b russia presses deeper into georgia  u s  says regime change is goal  b abhinav bindra wins first ever individual olympic gold medal for india  b  u s  ship heads for arctic to define territory  b drivers in a jerusalem taxi station threaten to quit rather than work for their 

In [1809]:
headlines=[]

for row in range(0,len(Data[index])):
    headlines.append(' '.join(str(x) for x in Data.iloc[row,0:25]))

In [1810]:
headlines[0]

'b georgia  downs two russian warplanes  as countries move to brink of war  b breaking  musharraf to be impeached   b russia today  columns of troops roll into south ossetia  footage from fighting  youtube   b russian tanks are moving towards the capital of south ossetia  which has reportedly been completely destroyed by georgian artillery fire  b afghan children raped with  impunity   u n  official says   this is sick  a three year old was raped and they do nothing  b     russian tanks have entered south ossetia whilst georgia shoots down two russian jets   b breaking  georgia invades south ossetia  russia warned it would intervene on so s side  b the  enemy combatent  trials are nothing but a sham  salim haman has been sentenced to       years  but will be kept longer anyway just because they feel like it   b georgian troops retreat from s  osettain capital  presumably leaving several hundred people killed   video   b did the u s  prep georgia for war with russia   b rice gives green

In [1811]:
Headlines=pd.DataFrame(headlines)

In [1812]:
Headlines=Headlines.astype('object')
Headlines.columns=['News']
Headlines.tail()

News
1981  barclays and rbs shares suspended from trading...
1982        scientists to australia  if you want to ...
1983  explosion at airport in istanbul yemeni former...
1984  jamaica proposes marijuana dispensers for tour...
1985  a     year old woman in mexico city finally re...

# Checking for any blanks values

In [1813]:
blanks= []

for i,News in Headlines.itertuples():
    if type(News)==str:
        if News.isspace():
            blanks.append(i)
print(len(blanks), 'blanks: ', blanks)

0 blanks:  []


In [1814]:
Headlines.drop(blanks, inplace=True)

In [1815]:
Headlines

News
0     b georgia  downs two russian warplanes  as cou...
1     b why wont america and nato help us  if they w...
2     b remember that adorable   year old who sang a...
3     b  u s  refuses israel weapons to attack iran ...
4     b all the experts admit that we should legalis...
...                                                 ...
1981  barclays and rbs shares suspended from trading...
1982        scientists to australia  if you want to ...
1983  explosion at airport in istanbul yemeni former...
1984  jamaica proposes marijuana dispensers for tour...
1985  a     year old woman in mexico city finally re...

[1986 rows x 1 columns]

# Training the model

In [1816]:
from sklearn.model_selection import train_test_split

In [1817]:
X=Headlines['News']
y=df['Label'].values

In [1818]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [1819]:
from sklearn.feature_extraction.text import TfidfVectorizer

In [1820]:
tfidf=TfidfVectorizer(ngram_range=(2,2))

In [1821]:
train_X=tfidf.fit_transform(X_train)

In [1822]:
test_X = tfidf.transform(X_test)   


# Random Forest

In [1823]:
from sklearn.ensemble import RandomForestClassifier

In [1932]:
randomclassifier=RandomForestClassifier(n_estimators=250,criterion='entropy')
randomclassifier.fit(train_X,y_train)

RandomForestClassifier(bootstrap=True, ccp_alpha=0.0, class_weight=None,
                       criterion='entropy', max_depth=None, max_features='auto',
                       max_leaf_nodes=None, max_samples=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=1, min_samples_split=2,
                       min_weight_fraction_leaf=0.0, n_estimators=250,
                       n_jobs=None, oob_score=False, random_state=None,
                       verbose=0, warm_start=False)

In [1844]:
predictions = randomclassifier.predict(test_X)
predictions_prob_train = randomclassifier.predict_proba(train_X)[:,1]
predictions_prob_test = randomclassifier.predict_proba(test_X)[:,1]

In [1845]:
from sklearn.metrics import confusion_matrix,classification_report,accuracy_score,roc_auc_score

In [1846]:
print("Confusion Matrix of Random Forest:",confusion_matrix(y_test,predictions))


Confusion Matrix of Random Forest: [[ 52 135]
 [ 50 161]]


In [1847]:
print('Classification Report: ',classification_report(y_test,predictions))


Classification Report:                precision    recall  f1-score   support

           0       0.51      0.28      0.36       187
           1       0.54      0.76      0.64       211

    accuracy                           0.54       398
   macro avg       0.53      0.52      0.50       398
weighted avg       0.53      0.54      0.51       398



In [1848]:
print('Accuracy Score',accuracy_score(y_test,predictions))

Accuracy Score 0.535175879396985


In [1933]:

print ("AUC of Test:", roc_auc_score(y_test,predictions_prob_test))

AUC of Test: 0.5230630813290418


# Decision Tree

In [1600]:
from sklearn.tree import DecisionTreeClassifier

In [1874]:
dtree=DecisionTreeClassifier(random_state=42,criterion='gini',min_samples_leaf=8,min_samples_split=24)

In [1875]:
dtree.fit(train_X,y_train)

DecisionTreeClassifier(ccp_alpha=0.0, class_weight=None, criterion='gini',
                       max_depth=None, max_features=None, max_leaf_nodes=None,
                       min_impurity_decrease=0.0, min_impurity_split=None,
                       min_samples_leaf=8, min_samples_split=24,
                       min_weight_fraction_leaf=0.0, presort='deprecated',
                       random_state=42, splitter='best')

In [1876]:
predictions=dtree.predict(test_X)
predictions_prob_train=dtree.predict_proba(train_X)[:,1]
predictions_prob_test=dtree.predict_proba(test_X)[:,1]

In [1883]:
print("AUC:",roc_auc_score(y_test,predictions_prob_test))
print("Accuracy Score",accuracy_score(y_test,predictions))

AUC: 0.5230630813290418
Accuracy Score 0.5326633165829145


In [1878]:
print('Classification Report: ',classification_report(y_test,predictions))

Classification Report:                precision    recall  f1-score   support

           0       0.50      0.53      0.52       187
           1       0.56      0.54      0.55       211

    accuracy                           0.53       398
   macro avg       0.53      0.53      0.53       398
weighted avg       0.53      0.53      0.53       398



In [1879]:
print("Confusion Matrix:",confusion_matrix(y_test,predictions))

Confusion Matrix: [[ 99  88]
 [ 98 113]]


# Naive Bayes

In [1908]:
from sklearn.naive_bayes import BernoulliNB

In [1909]:
bnb=BernoulliNB(alpha=.001)

In [1910]:
bnb.fit(train_X,y_train)

BernoulliNB(alpha=0.001, binarize=0.0, class_prior=None, fit_prior=True)

In [1918]:
predictions = bnb.predict(test_X)
predictions_prob = bnb.predict_proba(test_X)[:,1]

In [1919]:
print("Confusion matrix:",confusion_matrix(y_test,predictions))

Confusion matrix: [[ 52 135]
 [ 61 150]]


In [1921]:
print("AUC:",roc_auc_score(y_test,predictions_prob))

AUC: 0.4766454621486682


In [1922]:
print("accuracy score:",accuracy_score(y_test,predictions))

accuracy score: 0.507537688442211


In [1907]:
print('Classification Report: ',classification_report(y_test,predictions))

Classification Report:                precision    recall  f1-score   support

           0       0.46      0.28      0.35       187
           1       0.53      0.71      0.60       211

    accuracy                           0.51       398
   macro avg       0.49      0.49      0.48       398
weighted avg       0.50      0.51      0.48       398



# Boosting on Decision Tree

In [1711]:
from sklearn.ensemble import AdaBoostClassifier 


In [1712]:
adaboost=AdaBoostClassifier(dtree,n_estimators=200,random_state=42)

In [1713]:
adaboost.fit(train_X,y_train)

AdaBoostClassifier(algorithm='SAMME.R',
                   base_estimator=DecisionTreeClassifier(ccp_alpha=0.0,
                                                         class_weight=None,
                                                         criterion='gini',
                                                         max_depth=None,
                                                         max_features=None,
                                                         max_leaf_nodes=None,
                                                         min_impurity_decrease=0.0,
                                                         min_impurity_split=None,
                                                         min_samples_leaf=8,
                                                         min_samples_split=24,
                                                         min_weight_fraction_leaf=0.0,
                                                         presort='deprecated',
                      

In [1923]:
predictions=adaboost.predict(test_X)
predictions_prob=adaboost.predict_proba(test_X)[:,1]

In [1924]:
print('Classfication Report :',classification_report(y_test,predictions))

Classfication Report :               precision    recall  f1-score   support

           0       0.53      0.38      0.44       187
           1       0.56      0.71      0.63       211

    accuracy                           0.55       398
   macro avg       0.55      0.54      0.53       398
weighted avg       0.55      0.55      0.54       398



In [1925]:
print('confusion_matrix :',confusion_matrix(y_test,predictions))

confusion_matrix : [[ 71 116]
 [ 62 149]]


In [1926]:
print('roc_score :',roc_auc_score(y_test,predictions_prob))

roc_score : 0.552474846034924


In [1928]:
print('Accuracy Score :', accuracy_score(y_test,predictions))

Accuracy Score : 0.5527638190954773


# Boosting on Random forest

In [1760]:
adaboost_rtree=AdaBoostClassifier(randomclassifier,n_estimators=200,random_state=42)

In [1761]:
adaboost_rtree.fit(train_X,y_train)

AdaBoostClassifier(algorithm='SAMME.R',
                   base_estimator=RandomForestClassifier(bootstrap=True,
                                                         ccp_alpha=0.0,
                                                         class_weight=None,
                                                         criterion='gini',
                                                         max_depth=None,
                                                         max_features='auto',
                                                         max_leaf_nodes=None,
                                                         max_samples=None,
                                                         min_impurity_decrease=0.0,
                                                         min_impurity_split=None,
                                                         min_samples_leaf=1,
                                                         min_samples_split=2,
                                       

In [1929]:
predictions=adaboost_rtree.predict(test_X)
predictions_proba=adaboost_rtree.predict_proba(test_X)[:,1]
print('Classfication Report :',classification_report(y_test,predictions))
print('confusion_matrix :',confusion_matrix(y_test,predictions))
print('roc_score :',roc_auc_score(y_test,predictions_proba))
print('Accuracy Score',accuracy_score(y_test,predictions))

Classfication Report :               precision    recall  f1-score   support

           0       0.49      0.50      0.50       187
           1       0.55      0.54      0.54       211

    accuracy                           0.52       398
   macro avg       0.52      0.52      0.52       398
weighted avg       0.52      0.52      0.52       398

confusion_matrix : [[ 94  93]
 [ 98 113]]
roc_score : 0.5428947968674759
Accuracy Score 0.5201005025125628


# Voting Classifier

In [1942]:
from sklearn.ensemble import VotingClassifier

In [1943]:
vote_clf=VotingClassifier([('Random Forest',randomclassifier),('Decision Tree',dtree),('Adaboost Dtree',adaboost),
                           ('Adaboost RF',adaboost_rtree)],voting='hard')

In [1944]:
vote_clf.fit(train_X,y_train)

VotingClassifier(estimators=[('Random Forest',
                              RandomForestClassifier(bootstrap=True,
                                                     ccp_alpha=0.0,
                                                     class_weight=None,
                                                     criterion='entropy',
                                                     max_depth=None,
                                                     max_features='auto',
                                                     max_leaf_nodes=None,
                                                     max_samples=None,
                                                     min_impurity_decrease=0.0,
                                                     min_impurity_split=None,
                                                     min_samples_leaf=1,
                                                     min_samples_split=2,
                                                     min_weight_fraction_leaf

In [1945]:
predictions=vote_clf.predict(test_X)

print('roc_score :',roc_auc_score(y_test,predictions))
print('Accuracy Score',accuracy_score(y_test,predictions))


roc_score : 0.5485211749499455
Accuracy Score 0.5477386934673367


In [1946]:
print('Classfication Report :',classification_report(y_test,predictions))
print('confusion_matrix :',confusion_matrix(y_test,predictions))

Classfication Report :               precision    recall  f1-score   support

           0       0.52      0.56      0.54       187
           1       0.58      0.54      0.56       211

    accuracy                           0.55       398
   macro avg       0.55      0.55      0.55       398
weighted avg       0.55      0.55      0.55       398

confusion_matrix : [[105  82]
 [ 98 113]]


# As We clould see applying Adaboost on Random Forest gives the highest ROC score of 55%. 